pkg/fixtures/ddb_kv_writer.go

```diff
@@ -16,8 +16,18 @@ type KvstoreFixture struct {
 }
 
 type dynamoDbKeyValueFixtureWriter struct {
-	config cfg.Config
-	logger mon.Logger
+	config  cfg.Config
+	logger  mon.Logger
+	modelId *mdl.ModelId
+}
+
+func DynamoDbKvStoreFixtureWriterFactory(modelId *mdl.ModelId) FixtureWriterFactory {
+	return func(cfg cfg.Config, logger mon.Logger) FixtureWriter {
+		writer := NewDynamoDbKvStoreFixtureWriter(cfg, logger).(*dynamoDbKeyValueFixtureWriter)
```

Comment:
Instead of the type assertion we should change the return type of `NewDynamoDbKvStoreFixtureWriter`, and we can also make it private: `newDynamoDbKvStoreFixtureWriter `

Reply:
Same comment applies for all other writers

Reply:
if we did that, then we would not be able to use the cachedWriter. Still want that change?

Reply:
i just had a second look at it and it might be that we don't need the cache anymore, because each fixture writer would now be a different instance of the same struct due to the additional modelId/metadata that i added in this ticket.

Reply:
👍

Reply:
Makes sense. Kill the cache!